### ISSUE TITLE
Fails to parse record with unescaped parenthesis

### ISSUE THREAD

**Issue by marvin-yorke** [davedelong/CHCSVParser#79]
given the record

    16681;6;Orehovyj boulevard, ul. Musy Dzhalilja (odd side);20;out;55.6141571054;37.7460757208;800;34;34;0;0;0;0;0;1

library fails to parse the 3rd field with the following error:

    Unexpected delimiter. Expected ';' (0x3B), but got '(' (0x28)

Is there any way to parse this data without altering it (e.g adding quotes)?

**Comment by davedelong** [comment]
Thanks for reporting this. I added a unit test to parse the exact text you provided. and it seems to have no problem with it. I tried parsing it as the in-memory string, and as a file written to disk (which is similar to what [your code was doing](https://github.com/nextgis/metro4all-ios/commit/48b002e0365fc165ae1a4987f530f1c73df2d588#diff-a33e224f1b2f164a43c56fdde93e89c5L138)).  Both tests pass without modification to the parser, so I'm not sure what the issue is here.

**Comment by davedelong** [comment]
Are the URLs you're parsing remote (coming in over a network connection) or local file URLs? Do you have an example of either that I could try?

**Comment by marvin-yorke** [comment]
Hi Dave,
I'm downloading an archive from the server, unpack it into Documents directory and supply a URL to the file in Documents dir.
You can find the files I'm using in the following archive: http://metro4all.org/data/msk.zip
The file I've encountered the problem in is `portals_ru.csv`

**Comment by davedelong** [comment]
Thanks @marvin-yorke. I incorporated the `portals.csv` file into the unit tests, but they're still passing on my machine. 😕

**Comment by marvin-yorke** [comment]
Hm, ok, I've cloned the repo and run the tests and it works on my machine too. I should have mentioned that original case was observed on iOS, not OS X. Could this make any difference? Also I've installed the library from Cocoapods, not from github, although there's no major difference to the latest code.. Anyway, I'll try again with my iOS app and let you know about the results

**Comment by marvin-yorke** [comment]
I've checked the issue again and here's the line that breaks the parsing
`17530;2;"Крокус Экспо" (павильон 1, 2);215;both;55.8235522598;37.3855503584;800;56;0;0;0;400;950;23;0`
Turns out that it's not parentheses that cause the issue, but quotes. And now I'm not quite sure that it's a problem of the parser and not my data. What do you think?

**Comment by davedelong** [comment]
Yes, that is a problem with the data. When the parser encounters a field that starts with `"`, it assumes the field ends with the corresponding closing `"`. And then since the next character after the closing `"` isn't a delimiter (`;`), it aborts with an error.

**Comment by marvin-yorke** [comment]
Then could you please help me on how to correct my data?

**Comment by danieljfarrell** [comment]
The solutions seems pretty clear: don't start an field with quoted text; or if a field starts with quoted text wrap the whole field in a quotes.

**Comment by h3dkandi** [comment]
Is there a property that can turn of such behavior. Or some work around without me having to edit the file I am parsing.
